File: src/lessons/welcome/array/search/SecondMaxValueEntity.java
package lessons.welcome.array.search;

import static plm.core.ValueSerializer.*;

import plm.universe.bat.BatEntity;

public class SecondMaxValueEntity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(max2Value((int[])param[0])));
    }
  }

  /* BEGIN TEMPLATE */
  // computes the index of the second maximum of the values contained in tab variable
  public int max2Value(int[] tab)
  {
    /* BEGIN SOLUTION */
    int max = Integer.MIN_VALUE;
    int sec = Integer.MIN_VALUE;
    for (int i = 0; i < tab.length; i++)
      if (tab[i] > max) {
        sec = max;
        max = tab[i];
      } else if (tab[i] > sec) {
        sec = tab[i];
      }

    return sec;
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/welcome/array/search/SecondMaxValueEntity.py
from RemoteBat import *
def run():
    count = getTestCount()
    for i in range(count):
        param = deserialize(getTest(i))
        setTestResult(i, serialize(max2Value(param[0])))

# BEGIN TEMPLATE
def max2Value(nums):
# BEGIN SOLUTION
  max=-10000000
  sec=-10000000
  for i in range(len(nums)):
    if nums[i] > max:
      sec = max
      max = nums[i]
    elif nums[i] > sec:
      sec = nums[i]
  return sec
# END SOLUTION
# END TEMPLATE
